tests/loaders_test.py

```diff
@@ -500,3 +502,46 @@ def test_get_rule_file_hash_when_file_not_found():
     assert isinstance(hash, bytes)
     b64Hash = b64encode(hash).decode('ascii')
     assert 'zR1Ml8y8S8Z/I5j7b48OH+DJqUw=' == b64Hash
+
+
+def test_load_yaml_recursive_import():
+    config = {}
+    rules_loader = FileRulesLoader(config)
+
+    root_path = os.path.join(loaders_test_cases_path, 'recursive_import/root.yaml')
+    branch_path = os.path.join(loaders_test_cases_path, 'recursive_import/branch.yaml')
+    leaf_path = os.path.join(loaders_test_cases_path, 'recursive_import/leaf.yaml')
+
+    leaf_yaml = rules_loader.load_yaml(leaf_path)
+    assert leaf_yaml == {
+        'name': 'leaf',
+        'rule_file': leaf_path,
+    }
+    assert sorted(rules_loader.import_rules.keys()) == [
+        branch_path,
+        leaf_path,
+    ]
+    assert rules_loader.import_rules[branch_path] == [
+        root_path,
+    ]
+    assert rules_loader.import_rules[leaf_path] == [
+        branch_path,
+    ]
+
+    # reload the rule
+    leaf_yaml = rules_loader.load_yaml(leaf_path)
+    assert leaf_yaml == {
+        'name': 'leaf',
+        'rule_file': leaf_path,
+    }
+    assert sorted(rules_loader.import_rules.keys()) == [
+        branch_path,
+        leaf_path,
+    ]
+    assert rules_loader.import_rules[branch_path] == [
+        root_path,
+        root_path, # memory leak
```

Comment:
This is the memory leak.  Every time a rule gets loaded, only the top level entry is popped:

https://github.com/jertel/elastalert2/blob/1e72eb6c8f92753f2aa1d5357f850519bb72b465/elastalert/loaders.py#L244

The intermediary imports keep growing with every update.